CONTRIBUTING.md

```diff
@@ -71,35 +73,33 @@ type(scope): description
-
-Finally if your work is based on an issue on GitHub, please add in the body of the commit message `fix #1234` if it solves the issue #1234.
+Even though the scope is optional, we try to fill it in as it helps us better understand the impact of a change. We either use the name of the widget/connector/component impacted or we use impact topic (e.g. `docs`, `tooling`, `deps`, `ci`).
 
-Some examples of valid commit messages (first line):
+Finally, if your work is based on an issue on GitHub, please add in the body of the commit message "fix #1234" if it solves the issue #1234 (read "[Closing issues using keywords](https://help.github.com/en/articles/closing-issues-using-keywords)").
 
-- fix(searchbox): make magnifying glass bigger
-- chore(deps): update dependency rollup-plugin-babel to v3.0.7
-- fix(connectRefinementList): set default value for limit
-- chore: reword contributions guides
+Some examples of valid commit messages (used as first lines):
 
-## Branches organization
+> - fix(searchbox): increase magnifying glass size
+> - chore(deps): update dependency rollup-plugin-babel to v3.0.7
+> - fix(connectRefinementList): set default value for limit
+> - chore: reword contributions guides
 
-The project is based on the classic GitHub flow because we are building a library and each version must be crafted with care. We also maintain a branch specific for the older version of the library (currently v1). Finally, when building feature (that will go in the next minor version) we have specific branch.
+## Branch organization
 
-To sum up, we have:
+The project is based on the classic GitHub flow:
 
-- `develop` for the version -- Pull requests for bugs should be created against this branch
-- `feat/2.*` for the next minor version -- Pull requests for features should be created against this branch
-- `master` for the current stable version -- we usually don't make pull requests for this branch
-- `maintenance` for the previous major version (currently v1) -- Pull requests for critical bug fixes on the old version
+- `develop` for the current version being worked on – Pull requests for bugs and feature related to the current major version should be created against this branch
+- `master` for the latest stable version – we usually don't make pull requests on this branch
+- `vX` for each major version (`X` being a number) – Pull requests for critical bug fixes should be created against this branch
 
-You should do the dev and create pull requests according to the target version.
+Most of the time, your pull requests should target the `develop` branch, except if you're fixing a critical bug on an older version.
 
-_Note that no new features will be developed on the maintenance version._
+_Note that no new features will be developed on the `vX` branches._
```

Comment:
```suggestion
_Note that no new features will be developed or backported for the `vX` branches._
```

just a suggestion here to be a bit more clear in case someone disputes this